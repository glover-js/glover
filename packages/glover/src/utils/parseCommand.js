// @flow
import type { Command } from 'types/Command';
import type { Argv } from 'types/Yargs';

import CommandNotFoundError from './CommandNotFoundError';

const commands = {
  brew: 'brew',
  pick: 'pick',
  compose: 'compose',
};

export default function parseCommand(
  argv: Argv,
  rawArgv: Array<string>,
): Command {
  const [command] = argv._;
  if (commands[command]) {
    return commands[command];
  }
  if (!command) {
    return 'brew';
  }
  throw new CommandNotFoundError(rawArgv[2]);
}
