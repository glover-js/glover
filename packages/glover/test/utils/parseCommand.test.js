import parseCommand from '../../src/utils/parseCommand';

describe('Tests for parseCommand', () => {
  it('returns the parsed command from argv', () => {
    const argvA = { _: ['brew'] };
    const argvB = { _: ['pick'] };
    const argvC = { _: ['compose'] };
    expect(parseCommand(argvA)).toEqual('brew');
    expect(parseCommand(argvB)).toEqual('pick');
    expect(parseCommand(argvC)).toEqual('compose');
  });
  it('defaults to brew if it cannot parse the command from argv', () => {
    const argvA = { _: [''] };
    const argvB = { _: [] };
    expect(parseCommand(argvA)).toEqual('brew');
    expect(parseCommand(argvB)).toEqual('brew');
  });
  it('throws an error if the command is invalid', () => {
    const argv = { _: ['invalid-command'] };
    expect(() => parseCommand(argv)).toThrowErrorMatchingSnapshot();
  });
});
