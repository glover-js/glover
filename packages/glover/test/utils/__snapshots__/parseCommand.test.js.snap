// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for parseCommand throws an error if the command is invalid 1`] = `"Error: invalid-command is not a valid glover command"`;
